sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2648,6 +2648,13 @@ object GpuOverrides extends Logging {
         override def convertToGpu(child: Expression): GpuExpression =
           GpuMapEntries(child)
       }),
+    expr[StringToMap](
+      "Creates a map after splitting the input into key-value pairs using delimiters",
+      ExprChecks.projectOnly(TypeSig.MAP.nested(TypeSig.STRING), TypeSig.MAP.nested(TypeSig.STRING),
+        Seq(ParamCheck("str", TypeSig.STRING, TypeSig.STRING),
+          ParamCheck("pairDelim", TypeSig.lit(TypeEnum.STRING), TypeSig.STRING),
```

Comment:
nit: Actually it is a little hard to tell from the Spark code but it is also verifying that only literal values are used for the delims.

https://github.com/apache/spark/blob/4920014cc3a14019a2584c1059bc3d7b75426635/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala#L552

so you can have the Spark once marked as `TypeSig.lit(STRING)` also. This will let our docs be a bit more accurate.